pkg/block/azure/walker.go

```diff
@@ -104,6 +117,34 @@ func (a *BlobWalker) Walk(ctx context.Context, storageURI *url.URL, op block.Wal
 	return nil
 }
 
+// isBlobItemFolder returns true if the blob item is a folder.
+// Make sure that metadata is populated before calling this function.
+func isBlobItemFolder(blobItem *container.BlobItem) bool {
+	if blobItem.Metadata == nil {
+		return false
+	}
+	if blobItem.Properties.ContentLength == nil || *blobItem.Properties.ContentLength != 0 {
```

Comment:
Why do we need this check if we already have the `hdi_isfolder` metadata property?
Also `if blobItem.Properties.ContentLength == nil` - doesn't it mean that it might be actually a directory marker?

Reply:
ContentLength nil check is not an indicator - it should be populated. I've added a check or Int64Value to be on the safe side

Reply:
Can we perhaps change this condition to:
```suggestion
	if blobItem.Properties.ContentLength != nil && *blobItem.Properties.ContentLength != 0 {
```